programs/lib/flexpret_cond.c

```diff
@@ -0,0 +1,50 @@
+#include <flexpret.h>
+#include "flexpret_cond.h"
+
+fp_ret_t cond_wait(cond_t * cond) {
+    int hartid = read_hartid();
+    cond->waiting[hartid] = true;
+    lock_release(cond->lock);
+    while(cond->waiting[hartid]) {}
```

Comment:
not sure if `bool waiting[NUM_THREADS];` needs to be marked as `volatile` to keep the compiler from optimizing this loop out

Reply:
Good point. I am actually also not sure. But lets do it for good measure!